src/server/test-suites/transformers/test-suite-run-results.js

```diff
@@ -28,7 +28,25 @@ function getDuration({ created, taskLastUpdated }, hasResult) {
   return null
 }
 
-function transformTestSuiteRunResults(testRun) {
+function getStopTestEntity(testSuite, runId, runTaskStatus, showActions) {
```

Comment:
so this isn't the way to create entities. Entities live in - https://github.com/DEFRA/cdp-portal-frontend/blob/main/src/server/common/components/entity/template.njk

Creating it this way no one else can use it. I'd advise creating a generic action entity in the above file

Reply:
reworked this so the transformer conditionally creates either a button or empty text entities depending on the status etc